### DISCUSSION TITLE
Account Balance using swell-js

### DISCUSSION THREAD

**Discussion by igoraugustynski** [swellstores/community#356] [Help] [answered]
Hello,
I'm using `swell.client.account.get()` to get the account information of users. The "balance" property always shows 0 even if the user has some credit balance. Why is that?

**Comment by logeshswell** [comment, marked as answer]
I think it could be a gap in our implementation. But, I can suggest you with some workaround that would solve your problem.
* If your store is multi-currency enabled,
  * No change is required and balance will be fetched normally.
* If it's not enabled,
  * Pass your store default currency as options in `swell.init()` as shown below:
  ```ts
  const options = {
    currency: 'EUR'
  }
  swell.init('<store-id>', '<public_key>', options);
  ```

**Reply by igoraugustynski** [reply]
I can confirm that it works correctly when setting the currency. Thank You!
Another question -> How can I use this balance as a payment option during checkout? There is a mention about this in the documentation but no implementation details.

**Reply by logeshswell** [reply]
> How can I use this balance as a payment option during checkout?

You can make the below cart update request and it will include the available account balance to the cart, provided the customer is logged into the current cart session.

```ts
// Returns the list of payment methods available in the swell store.
const paymentMethods = await swell.settings.payments();

await swell.cart.update({
  billing: {
    method: 'account'
  }
});
```

If you need flexibility in setting the account balance value to be deduced in the checkout, then you need to use backend API to update the field `account_credit_amount`

**Reply by logeshswell** [reply]
If the cart_total_value <= account_balance, then you can also update the other billing details to place an order.

If cart_total_value > account_balance, then making the cart update request with an empty body should also include the available account balance to the cart and update the due amount in `capture_total` field.

You should use the other alternate payment methods like card, etc to complete the due payment by setting the corresponding `billing.method: 'card'`.
